packages/entity/src/rules/internal/PrivacyPolicyRuleOrdering.ts

```diff
@@ -0,0 +1,70 @@
+import invariant from 'invariant';
+
+import { RuleEvaluationResultType } from '../PrivacyPolicyRuleEnums';
+
+/**
+ * Privacy policy list of PrivacyPolicyRule can be reordered safely according to
+ * the following constraints on RuleEvaluationResultType:
+ * 1. Two adjacent ALLOW_OR_SKIP rules can be swapped.
+ * 2. Two adjacent DENY_OR_SKIP rules can be swapped.
+ * 3. SKIP rules can be swapped with any other adjacent rule.
+ *
+ * @param rulesetOriginalOrder - ruleset with original ordering defined in code
+ * @returns reordered ruleset depending on constraints above
+ */
+export function reorderRulesByRuleComplexityGroups<
```

Comment:
That could definitely work, too. Here's some thoughts:
- I think the main issue I see is that we don't have a way of enforcing that an actual unit test is run (that'd live in the application). That being said, we could just do the validation at evaluation time (validate that it is in the optimal order) and throw an invariant error if it isn't (not a runtime error). I somewhat like this idea. I also like the easier-to-debug nature of it.
- Alternatively, going off that idea, this could live entirely in the application layer and not be in the framework at all. Theoretically we could just have a lint rule to require all privacy rules have a complexity field and then have a second lint rule that analyzes their ordering.
- The natural extension of this is to calculate complexity at runtime, yep. Or even use some other heuristic to order them. In other similar frameworks, I've seen it even reordered according to which rule is most-often the deciding rule.

Reply:
Summarizing our discussion slack:
- For an application that would only ever hard code the complexity, I like the lint rule idea. It's unobtrusive and allows for easier debugging at runtime
- In the long term if we wanted to make a dynamic `complexity` system (ie) optional concept/field called “PrivacyPolicyOrderer” that had the ability to reorder, used postgres stats to update complexity every now and then, what you have already is probably fine